ekncontent/ekncontent/eknc-utils.c

```diff
@@ -203,6 +203,31 @@ eknc_get_ekn_version (const gchar *app_id,
   return g_strdup (stripped);
 }
 
+/**
+ * eknc_get_current_language:
+ *
+ * Gets the xapian friendly version of the current system language, or NULL if
+ * none set.
+ *
+ * Returns: (transfer none): the language
+ */
+const gchar *
+eknc_get_current_language (void)
+{
+  gchar **langs = (gchar **) g_get_language_names ();
```

Comment:
Don't cast this to `gchar **`; it's too easy to fool functions into modifying the strings, and they should not be modified. Instead cast it on a case-by-case basis when using functions such as `g_strv_length()` that don't correctly have `const` in their signature.